### ISSUE TITLE
[SCREENREADER]: Keyboard and screen reader focus MUST be trapped in modal windows

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#9229]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️  Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Right now our modal windows do a great job of trapping keyboard focus, but they sometimes let screen readers' virtual cursor get behind the dark screen, onto page content. This is not what we want, and should be trapped in the modal window. Screenshot attached below.

This issue needs to be addressed in our modal window component, and has been logged in https://github.com/department-of-veterans-affairs/va.gov-team/issues/1449. I'll mark this new issue blocked by that one, so it can be iceboxed until we're ready to move forward.

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/1449

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] Current keyboard focus behavior (trapped by pressing `TAB` or `SHIFT+TAB` is preserved
- [ ] Screen reader virtual cursors cannot go forward or backward out of the modal dialog

## Screenshots or Trace Logs

![Screen Shot 2020-05-18 at 2 46 03 PM](https://user-images.githubusercontent.com/934879/82262921-7261ae80-9927-11ea-96af-9ed31663ddc2.png)
